runtime/interp.c

```diff
@@ -1418,7 +1418,7 @@ do_resume: {
 #ifndef THREADED_CODE
     default:
 #ifdef _MSC_VER
-      __assume(0);
+      unreachable();
```

Comment:
Looking at this again, I think there is no reason to keep the `_MSC_VER` conditional, and that it should `caml_fatal_error` in all cases - in case of error (corrupted bytecode file) this would be more helpful to the user.

Reply:
There's a possibility that adding the `unreachable` annotation here instead of an error case would allow the compiler to optimize jumps (because MSVC doesn't support labels-as-values/computed gotos), but I'd need to experiment and benchmark to make sure.

Reply:
In the meantime I'm tempted to leave this as-is an revisit later, if that's fine.

Reply:
That's fine.